src/class_sqlThread.py

```diff
@@ -29,22 +18,144 @@
+            else:
+                sys.stderr.write(
+                    'ERROR trying to upgrade database. Error message: %s\n' % str(err))
+                os._exit(0)
+
+    @property
+    def sql_schema_version(self):
+        self._current_level = self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self, cur, conn, current_level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (current_level + 1,)
+        self.cur.execute(item, parameters)
+        self._current_level = self.__get_current_settings_version()
+
+    # @sql_schema_version.setter
+    def upgrade_to_latest(self, cur, conn, current_level=None):
+        """
+            Initialise upgrade level
+        """
+
+        # Declare variables
+        self.conn = conn
+        self.cur = cur
+        if current_level:
+            self._current_level = current_level
+        else:
+            self._current_level = self.__get_current_settings_version()
+        self.max_level = 11
+
+        # call upgrading level in loop
+        for l in range(self._current_level, self.max_level):
+            if int(l) == 1:
+                continue
+            self._upgrade_one_level_sql_statement(l)
+            # self.increment_settings_version(l)
+            self._current_level = l
+
+    def increment_settings_version(self, level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (level + 1,)
+        self.cur.execute(item, parameters)
+
+
+class sqlThread(threading.Thread, UpgradeDB):
     """A thread for all SQL operations"""
 
     def __init__(self):
+        super(sqlThread, self).__init__()
         threading.Thread.__init__(self, name="SQL")
+        self.__flag = threading.Event()
+
+        # Adding for pause/resume thread
+        self.__flag = threading.Event()
+        self.__flag.set()  # Set to True
+        self.__running = threading.Event()  # Used to stop the thread identification
+        self.__running.set()  # Set running to True
+
+    def pause(self):
+        self.__flag.clear()  # Set to False to block the thread
+
+    def resume(self):
+        self.__flag.set()  # Set to True, let the thread stop blocking
 
-    def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
+    def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements,
+        # Redefinition-of-parameters-type-from-tuple-to-str, R0204, line-too-long, E501
         """Process SQL queries from `.helper_sql.sqlSubmitQueue`"""
         helper_sql.sql_available = True
-        self.conn = sqlite3.connect(state.appdata + 'messages.dat')
-        self.conn.text_factory = str
-        self.cur = self.conn.cursor()
+        self.conn, self.cur = connection_build()
```

Comment:
this should be in parent class

Reply:
Inherit connection vars from UpgradeDB class

Reply:
Moved in to ```__init__```